lib/ui/page/swap_detail.dart

```diff
@@ -0,0 +1,262 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:mixswap_sdk_dart/mixswap_sdk_dart.dart';
+
+import '../../db/mixin_database.dart';
+import '../../service/mix_swap.dart';
+import '../../util/extension/extension.dart';
+import '../../util/hook.dart';
+import '../../util/l10n.dart';
+import '../../util/logger.dart';
+import '../../util/r.dart';
+import '../widget/brightness_observer.dart';
+import '../widget/buttons.dart';
+import '../widget/mixin_appbar.dart';
+import '../widget/symbol.dart';
+import '../widget/transaction_info_tile.dart';
+
+class SwapDetail extends HookWidget {
+  const SwapDetail({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final pathParameters = context.pathParameters;
+    final traceId = pathParameters['id'];
+    final queryParams = context.queryParameters;
+    final sourceId = useState(queryParams['source']);
+    final destId = useState(queryParams['dest']);
+
+    if (traceId == null) {
+      return const SizedBox();
+    }
+
+    final order = useState<Order?>(null);
+    useEffect(() {
+      var canceled = false;
+      scheduleMicrotask(() async {
+        while (!canceled) {
+          try {
+            final traceOrder = (await MixSwap.client.getOrder(traceId)).data;
+            sourceId.value = traceOrder.payAssetUuid;
+            destId.value = traceOrder.receiveAssetUuid;
```

Comment:
Would you extract code snippet under from condition to be a hook widget?

Reply:
I've extracted this part as a function for now.